core/payment/src/dao/payment.rs

```diff
@@ -30,97 +27,83 @@ impl<'c> AsDao<'c> for PaymentDao<'c> {
     }
 }
 
-// FIXME: This could probably be a function
-macro_rules! query {
-    () => {
-        dsl::pay_payment
-            .inner_join(
-                agreement_dsl::pay_agreement.on(dsl::owner_id
-                    .eq(agreement_dsl::owner_id)
-                    .and(dsl::agreement_id.eq(agreement_dsl::id))),
-            )
-            .select((
-                dsl::id,
-                dsl::owner_id,
-                dsl::role,
-                dsl::agreement_id,
-                dsl::allocation_id,
-                dsl::amount,
-                dsl::timestamp,
-                dsl::details,
-                agreement_dsl::peer_id,
-                agreement_dsl::payee_addr,
-                agreement_dsl::payer_addr,
-            ))
-    };
-}
-
 impl<'c> PaymentDao<'c> {
     async fn insert(
         &self,
         payment: WriteObj,
-        debit_note_ids: Vec<String>,
-        invoice_ids: Vec<String>,
+        activity_payments: Vec<ActivityPayment>,
```

Comment:
It only looks like that because of `cargo fmt`. ;) If you skip the comments and log::trace lines it's just: 
* 2 loops
* 1 if
* 6 assignments
* 6 statements

If you still thinks this is too much I can move the loops for inserting activity payments and agreement payments into separate functions.

Reply:
Done.